sdk/mgmtcommon/AppAuthentication/Azure.Services.AppAuthentication/AzureServiceTokenProvider.cs

```diff
@@ -44,7 +44,7 @@ public class AzureServiceTokenProvider
         /// KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
         /// </code>
         /// </example>
-        public TokenCallback KeyVaultTokenCallback => async (authority, resource, scope) =>
+        public virtual TokenCallback KeyVaultTokenCallback => async (authority, resource, scope) =>
```

Comment:
Have we decided to use virtual vs interface to support mocks?  Tradeoffs either way, just want to be sure virtual is consistent across the rest of the SDK.

Reply:
virtual methods seem to be the standard adopted by most of the azure SDK components.